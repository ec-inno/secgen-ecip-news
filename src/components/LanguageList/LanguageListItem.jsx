import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import { Link } from 'gatsby';

import Context from '../../Context';

// As close as possible to https://github.com/ec-europa/europa-component-library/blob/v2-dev/src/systems/ec/implementations/react/page-structure/language-list/src/LanguageListItem.jsx
const LanguageListItem = ({ label, lang, href, isActive, isOverlay }) => {
  const { dispatch } = useContext(Context);

  return (
    <Link
      onClick={() =>
        dispatch({
          type: 'SET_LANGUAGE_OVERLAY_VISIBILITY',
          hideOverlay: true,
        })
      }
      lang={lang}
      hrefLang={lang}
      to={href}
      className="ecl-language-list__link ecl-link ecl-link--standalone"
      {...(isActive && {
        icon: {
          shape: 'ui--check',
          size: 'xs',
        },
      })}
      {...(lang && {
        lang,
        hrefLang: lang,
        ...(isOverlay ? { rel: 'alternate' } : {}),
      })}
    >
      {label}
    </Link>
  );
};

LanguageListItem.propTypes = {
  label: PropTypes.string,
  lang: PropTypes.string,
  href: PropTypes.string,
  isActive: PropTypes.bool,
  isOverlay: PropTypes.bool,
};

LanguageListItem.defaultProps = {
  label: '',
  lang: '',
  href: '',
  isActive: false,
  isOverlay: false,
};

export default LanguageListItem;
