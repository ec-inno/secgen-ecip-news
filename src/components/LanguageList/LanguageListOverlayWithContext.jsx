import React, { useContext } from 'react';

import Context, { SET_LANGUAGE_OVERLAY_VISIBILITY } from '../../Context';

import LanguageListOverlay from './LanguageListOverlay';

const LanguageListOverlayWithContext = props => {
  const { dispatch, store } = useContext(Context);
  const { hideOverlay } = store;

  const closeHandler = () =>
    dispatch({ type: SET_LANGUAGE_OVERLAY_VISIBILITY, hideOverlay: true });

  return (
    <LanguageListOverlay
      hidden={hideOverlay}
      closeHandler={closeHandler}
      {...props}
    />
  );
};

export default LanguageListOverlayWithContext;
