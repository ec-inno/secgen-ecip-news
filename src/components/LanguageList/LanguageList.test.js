import React from 'react';
import renderer from 'react-test-renderer';
import LanguageList from './LanguageList';

describe('LanguageList', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LanguageList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
