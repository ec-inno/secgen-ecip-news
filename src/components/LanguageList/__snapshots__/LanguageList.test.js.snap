// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageList renders correctly 1`] = `
<div
  className="ecl-row"
>
  <div
    className="ecl-language-list__column ecl-col-12 ecl-col-lg-4 ecl-offset-lg-2"
  >
    <ul
      className="ecl-language-list__list"
    />
  </div>
  <div
    className="ecl-language-list__column ecl-col-12 ecl-col-lg-4"
  >
    <ul
      className="ecl-language-list__list"
    />
  </div>
</div>
`;
